packages/blade/src/components/Checkbox/CheckboxIcon/CheckboxIconWrapperStyles.ts

```diff
@@ -0,0 +1,73 @@
+import type { CSSObject } from 'styled-components';
+import type { CheckboxIconProps } from './CheckboxIcon';
+import type { Theme } from '~components/BladeProvider';
+import { getIn, makeSize, makeSpace } from '~utils';
+
+const variants = {
+  default: {
+    border: {
+      checked: 'colors.brand.primary.500',
+      unchecked: 'colors.brand.gray.400',
+    },
+    background: {
+      checked: 'colors.brand.primary.500',
+      unchecked: 'colors.brand.gray.200',
+    },
+  },
+  disabled: {
+    border: {
+      checked: 'colors.brand.gray.300',
+      unchecked: 'colors.brand.gray.300',
+    },
+    background: {
+      checked: 'colors.brand.gray.300',
+      unchecked: 'colors.brand.gray.300',
+    },
+  },
+  negative: {
+    border: {
+      checked: 'colors.feedback.border.negative.highContrast',
+      unchecked: 'colors.feedback.border.negative.highContrast',
+    },
+    background: {
+      checked: 'colors.feedback.background.negative.highContrast',
+      unchecked: 'colors.feedback.background.negative.lowContrast',
+    },
+  },
+};
+
+export type CheckboxRectProps = Omit<CheckboxIconProps, 'state'> & {
+  isChecked: boolean;
+};
+
+const getCheckboxIconWrapperStyles = ({
+  theme,
+  isChecked,
+  isDisabled,
+  isNegative,
+}: CheckboxRectProps & { theme: Theme }): CSSObject => {
+  let variant: 'default' | 'disabled' | 'negative' = 'default';
+  if (isDisabled) variant = 'disabled';
+  if (isNegative) variant = 'negative';
+  const checked = isChecked ? 'checked' : 'unchecked';
+  const backgroundColor = variants[variant].background[checked];
+  const borderColor = variants[variant].border[checked];
+
+  return {
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+    width: '16px',
+    height: '16px',
+    borderWidth: '1px',
+    borderStyle: 'solid',
+    margin: '0px',
+    marginTop: '3px',
```

Comment:
This seems like a peculiar value, 3px. Can't we utilize tokens here?

Reply:
Hmm yeah not really. We don't have token for 3px. We can get it close to 4px. But my OCD hits hard seeing a 1px unaligned icon.

Reply:
Is this on design side as well?

Reply:
Is this not a tokenizable spacing. It is just a discrepancy between how web and figma handle spacing. 

For example: In figma saurav made the checkbox have padding all around thus it looks okay.
But in code side we have to add margin since adding padding will mess up the size of icon because of the `border-box` rules of CSS.